Sources/Features/Client/BacktraceReporter.swift

```diff
@@ -96,3 +103,86 @@ extension BacktraceReporter {
         return resource
     }
 }
+
+#if os(iOS) || os(tvOS)
+typealias Application = UIApplication
+#elseif os(macOS)
+typealias Application = NSApplication
+#else
+#error("Unsupported platform")
+#endif
+
+//// Provides notification interfaces for BacktraceOOMWatcher and Breadcrumbs support
+extension BacktraceReporter {
+    internal func enableOomWatcher() {
+        self.backtraceOomWatcher.start()
+        
+        NotificationCenter.default.addObserver(self,
+                                               selector: #selector(handleTermination),
+                                               name: Application.willTerminateNotification,
+                                               object: nil)
+        
+        NotificationCenter.default.addObserver(self,
+                                               selector: #selector(didBecomeActiveNotification),
+                                               name: Application.didBecomeActiveNotification, object: nil)
+        
+        NotificationCenter.default.addObserver(self,
+                                               selector: #selector(willResignActiveNotification),
+                                               name: Application.willResignActiveNotification,
+                                               object: nil)
+        
+        #if os(iOS) || os(tvOS)
+        NotificationCenter.default.addObserver(self,
+                                               selector: #selector(applicationWillEnterForeground),
+                                               name: Application.willEnterForegroundNotification,
+                                               object: nil)
+        
+        NotificationCenter.default.addObserver(self,
+                                               selector: #selector(didEnterBackgroundNotification),
+                                               name: Application.didEnterBackgroundNotification,
+                                               object: nil)
+        
+        NotificationCenter.default.addObserver(self,
+                                               selector: #selector(handleLowMemoryWarning),
+                                               name: Application.didReceiveMemoryWarningNotification,
+                                               object: nil)
+        #endif
+        
+        #if os(macOS)
+        let source = DispatchSource.makeMemoryPressureSource(eventMask: [.critical, .warning], queue: nil)
+        let dispatchQueue = DispatchQueue.init(label: "Backtrace")
+        dispatchQueue.async {
+            source.setEventHandler {
+                if source.mask == DispatchSource.MemoryPressureEvent.warning ||
+                    source.mask == DispatchSource.MemoryPressureEvent.critical {
+                    self.handleLowMemoryWarning()
```

Comment:
No action required. I understand that `Backtrace OOM` lives forever once initialized therefore it does not matter much. However, it's a good practice to use weak references within closures. 

```
      closure { [weak self] in 
           guard let self = self else { return } 
           self.handleLowMemoryWarning()
       }
```